src/components/Preferences/LocalIdentities.js

```diff
@@ -456,10 +462,17 @@ const LocalIdentities = React.memo(function LocalIdentities({
     }
   }, [identities, dao, addressesSelected, someSelected])
 
-  const handleClearAll = useCallback(() => {
+  const handleRemoveSelected = useCallback(() => {
     setConfirmationModalOpened(false)
-    onClearAll()
-  }, [onClearAll])
+    const selectedAddresses = Array.from(addressesSelected.entries()).reduce(
+      (array, [address, selected]) => [
+        ...array,
+        ...(selected ? [address] : []),
+      ],
+      []
+    )
```

Comment:
@AquiGorka Would `addressesSelected.entries().filter(([_, selected]) => selected)` work just as well in this case?

Reply:
I do think it would :)